ui/src/app/(index)/page.tsx

```diff
@@ -1,7 +1,79 @@
-const Home = () => {
+import Image from 'next/image';
+import BeforeLogin from '@/components/home/BeforeLogin';
+import AfterLogin from '@/components/home/AfterLogin';
+import {getCurrentUser} from '@/access/AuthenticationService';
+import Role from '@/access/Role';
+import LoginButton from '@/components/home/LoginButton';
+import {getAnnouncements, getNumberOfGroupings, getNumberOfMemberships} from '@/services/GroupingsApiService';
+import React from 'react';
+import Announcement from '@/components/home/Announcement';
+
+const Home = async () => {
+    const [currentUser, numberOfGroupings, numberOfMemberships, announcements] = await Promise.all([
+        getCurrentUser(),
+        getNumberOfGroupings(),
+        getNumberOfMemberships(),
+        getAnnouncements()
+    ]);
+
+    const handleAnnouncements = () => {
+        if (Object.prototype.hasOwnProperty.call(announcements, 'announcements')) {
+            return announcements.announcements
+                .filter((announcement) => announcement.state !== 'Active')
+                .map((announcement) => announcement.message);
+        } else {
+            return [];
+        }
+    };
+
     return (
-        null
+        <main>
+            <div className="container mt-5 mb-5">
+                {handleAnnouncements().map((announcement: string, index: number) => (
+                    <div key={index}>
+                        <Announcement announcement={announcement}/>
+                    </div>
```

Comment:
The `div` wrapping the `Announcement.tsx` can be removed and you should be able to put the `key` attribute on `Announcement.tsx`

Reply:
done